tools/misbehaving-jmx-server/src/main/java/org/datadog/supervisor/App.java

```diff
@@ -0,0 +1,165 @@
+package org.datadog.supervisor;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.ConnectException;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import org.datadog.Defaults;
+
+import lombok.extern.slf4j.Slf4j;
+
+import io.javalin.*;
+
+class AppConfig {
+    public int controlHttpPort;
+    public String rmiHostname;
+
+    public AppConfig() {
+        String controlHttpPortStr = System.getenv("SUPERVISOR_PORT");
+        this.rmiHostname = null;
+        if (controlHttpPortStr != null) {
+            this.controlHttpPort = Integer.parseInt(controlHttpPortStr);
+        } else {
+            this.controlHttpPort = 8088;
+        }
+    }
+}
+
+@Slf4j
+public class App {
+    private static Process process = null;
+    // marked when OS process is started
+    private static AtomicBoolean running = new AtomicBoolean(false);
+    // marked when the JMX server indicates that its ready
+    private static AtomicBoolean started = new AtomicBoolean(false);
+    private static final String jmxServerEntrypoint = "org.datadog.misbehavingjmxserver.App";
+    private static String selfJarPath;
+    private static AppConfig config;
+
+    public static void main(String[] args) throws IOException {
+        App.config = new AppConfig();
+
+        try {
+            selfJarPath = new File(App.class.getProtectionDomain().getCodeSource().getLocation().toURI().getPath())
+                    .getAbsolutePath();
+        } catch (Exception e) {
+            log.warn("Could not get self jar path");
+            e.printStackTrace();
+            System.exit(1);
+        }
+
+        Javalin app = Javalin.create();
+
+        app.post("/config/", ctx -> {
+            if (ctx.formParam("rmiHostname") != null) {
+                App.config.rmiHostname = ctx.formParam("rmiHostname");
+                log.info("Setting RMI Hostname to {}. Restarting now...", App.config.rmiHostname);
+                stopJMXServer();
+                try {
+                    startJMXServer();
+                    ctx.result("Process restarted").status(200);
+                } catch (Exception e) {
+                    ctx.result("Error restarting process: " + e.getMessage()).status(500);
+                }
+            }
+        });
+        app.get("/ready", ctx -> {
+            if (started.get()) {
+                ctx.status(200);
+            } else {
+                ctx.status(500);
+            }
+        });
+
+        log.info("Supervisor HTTP control interface at :{}", App.config.controlHttpPort);
+        app.start(App.config.controlHttpPort);
+
+        log.info("Starting JMX subprocess");
```

Comment:
Consider _not_ starting the jmx subprocess until we recieve our initial bundle of init values (ie, rmiHostname).

This is currently attempting to support a case where some users of `misbehaving-jmx-server` may need the custom rmi hostname and some may not, but in reality all of our users _do_ need this support, so lets bake that assumption in and not try to be too smart